app/src/docs/v1.api-spec.yaml

```diff
@@ -0,0 +1,2592 @@
+---
+openapi: 3.0.2
+info:
+  version: 1.0.0
+  title: NR PermitConnect Navigator Service (PCNS)
+  description: >-
+    A case management application meant to serve the needs of the NRM Permitting
+    Solutions Branch.
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+servers:
+  - url: /api/v1
+    description: This server
+security:
+  - BearerAuth: []
+    OpenID: []
+tags:
+  - name: Activity
+    description: Operations for managing Activities
+  - name: Document
+    description: Operations for managing Documents
+  - name: Enquiry
+    description: Operations for managing Enquiries
+  - name: Note
+    description: Operations for managing Notes
+  - name: Permit
+    description: Operations for managing Permits
+  - name: Roadmap
+    description: Operations for validating Roadmaps
+  - name: Submission
+    description: Operations for managing Submissions, Drafts, and Statistics
+paths:
+  /activity/validate/{activityId}:
+    get:
+      summary: Validates an Activity ID
+      description: Validates the format and existence of an Activity ID.
+      operationId: validateActivityId
+      tags:
+        - Activity
+      parameters:
+        - $ref: '#/components/parameters/Path-ActivityId'
+      responses:
+        '200':
+          description: The validation result
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  valid:
+                    type: boolean
+                    description: Whether the Activity ID is valid and exists
+                    example: true
+        '400':
+          description: Invalid Activity ID format
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Invalid activity Id format
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        default:
+          $ref: '#/components/responses/Error'
+  /document/:
+    put:
+      summary: Create or update a document
```

Comment:
I believe this api can only create the document, there is no update. Delete and re-upload is the only way.